core/src/commonMain/kotlin/com/xebia/functional/xef/textsplitters/TokenTextSplitter.kt

```diff
@@ -14,14 +15,12 @@ private class TokenTextSplitterImpl(
 
   override suspend fun splitText(text: String): List<String> {
     val inputIds = tokenizer.encode(text)
-    val stepSize = chunkSize - chunkOverlap
-
-    return inputIds.indices
-      .asSequence()
-      .filter { it % stepSize == 0 }
-      .map { startIdx -> inputIds.subList(startIdx, minOf(startIdx + chunkSize, inputIds.size)) }
-      .map { chunkIds -> tokenizer.decode(chunkIds) }
-      .toList()
+    fun decodeSegment(startIdx: Int): String {
+      val end = minOf(startIdx + chunkSize, inputIds.size)
+      return tokenizer.decode(inputIds.subList(startIdx, end))
+    }
+    val segments = IntProgression.fromClosedRange(0, inputIds.size - 1, chunkSize - chunkOverlap)
```

Comment:
Nit: at the cost of 1 additional allocation, we get slightly better readability? 🤔
 
```suggestion
    val segments = inputIds.indices step (chunkSize - chunkOverlap)
```